src/core/lib/gpr/time.cc

```diff
@@ -77,16 +77,19 @@ static gpr_timespec to_seconds_from_sub_second_time(int64_t time_in_units,
   } else if (time_in_units == INT64_MIN) {
     out = gpr_inf_past(type);
   } else {
-    if (time_in_units >= 0) {
-      out.tv_sec = time_in_units / units_per_sec;
-    } else {
-      out.tv_sec = (-((units_per_sec - 1) - (time_in_units + units_per_sec)) /
-                    units_per_sec) -
-                   1;
-    }
+    out.tv_sec = time_in_units / units_per_sec;
+    /// Our current usage does not seem to cause precision issues. Division for
```

Comment:
I don't understand why this comment was added

Reply:
My understanding is we should treat every number in the chain of operations `(time_in_units - (out.tv_sec * units_per_sec)) * (GPR_NS_PER_SEC / units_per_sec)` as floating numbers to avoid losing precision from division with remainders. I did not do that since current usage does not seem to cause that.

Reply:
Oh I think we're ok here because `GPR_NS_PER_SEC % units_per_sec == 0` always... let's drop in a `GPR_DEBUG_ASSERT` to verify that.